server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMessagesCreationProcessor.java

```diff
@@ -290,21 +300,40 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
         return new ValueWithId.MessageWithId(entry.getCreationId(), jmapMessage);
     }
 
-    private void assertUserIsSender(MailboxSession session, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
+    @VisibleForTesting
+    void assertUserIsSender(Username connectedUser, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
         if (!from.flatMap(DraftEmailer::getEmail)
-                .filter(email -> session.getUser().equals(Username.of(email)))
+                .filter(email -> emailBelongToTheConnectedUser(connectedUser, Username.of(email)))
                 .isPresent()) {
-            String allowedSender = session.getUser().asString();
+            String allowedSender = connectedUser.asString();
             throw new MailboxSendingNotAllowedException(allowedSender);
         }
     }
 
+    private boolean emailBelongToTheConnectedUser(Username connectedUser, Username fromUser) {
+        try {
+            return connectedUser.equals(fromUser) || emailIsAnAliasOfTheConnectedUser(connectedUser, fromUser);
+        } catch (RecipientRewriteTableException | RecipientRewriteTable.ErrorMappingException e) {
+            return false;
+        }
+    }
+
+    private boolean emailIsAnAliasOfTheConnectedUser(Username connectedUser, Username fromUser) throws RecipientRewriteTable.ErrorMappingException, RecipientRewriteTableException {
+        return connectedUser.getDomainPart().isPresent() &&
```

Comment:
I tried with `map` beforehand but the value is not used and we then need to use a `Throwing.function`. So instead of making the code cleaner it add some boilerplate IMHO

Reply:
ok